download-latest.sh

```diff
@@ -82,45 +45,11 @@ failure_usage() {
     echo 'Follow the steps on the docs: https://docs.meilisearch.com/advanced_guides/binary.html#how-to-compile-meilisearch'
 }
 
-# OS DETECTION
-echo 'Detecting OS distribution...'
-os_name=$(uname -s)
-if [ "$os_name" != "Darwin" ]; then
-    os_name=$(cat /etc/os-release | grep '^ID=' | tr -d '"' | cut -d '=' -f 2)
-fi
-echo "OS distribution detected: $os_name"
-case "$os_name" in
-'Darwin')
-    os='macos'
-    ;;
-'ubuntu' | 'debian')
-    os='linux'
-    ;;
-*)
-    failure_usage
-    exit 1
-esac
-
-# GET LATEST VERSION
-tags=$(curl -s 'https://api.github.com/repos/meilisearch/MeiliSearch/tags' \
-    | grep "$GREP_SEMVER_REGEXP" \
-    | grep 'name' \
-    | tr -d '"' | tr -d ',' | cut -d 'v' -f 2)
-
-latest=""
-for tag in $tags; do
-    if [ "$latest" = "" ]; then
-        latest="$tag"
-    else
-        semverLT $tag $latest
-        if [ $? -eq 1 ]; then
-            latest="$tag"
-        fi
-    fi
-done
+# MAIN
+os="$(get_os)"
+latest="$(sh .github/get-latest-stable-release.sh)"
```

Comment:
However I don't understanf why you split the scripts into two parts, will that always works using the curl command?
This script depends on a script located at a path, but the user doesn't have this path...

Reply:
I thought it would work because the user does:
```bash
$ curl -L https://install.meilisearch.com | sh
```
and `https://install.meilisearch.com` is a redirection to `https://raw.githubusercontent.com/meilisearch/MeiliSearch/master/download-latest.sh` (so to the GitHub repository where there are the other scripts)

Reply:
And I split it because I use the `get-latest-stable-release.sh` in another script `is-latest-release.sh` used in GHA. I did not want to duplicate the code.

Reply:
I understand why you split it, but the line I commented does not work now, this will try to run `sh` with a script located at `.github/get-latest-stable-release.sh` which doesn't exist at all for the user. The script is no more self contained now.

Reply:
Shit, so we have to duplicate the code?

Reply:
I think this is the best thing that we can do here, sadly.

Reply:
Snif! Changing that!